preview/Win7Msix/Win7MSIXInstaller/MsixRequest.cpp

```diff
@@ -218,4 +227,19 @@ void MsixRequest::SetUI(UI * ui)
 void MsixRequest::SetPackageInfo(PackageInfo* packageInfo) 
 {
     m_packageInfo = packageInfo; 
+}
+
+HRESULT MsixRequest::UnInstallPackage(std::wstring packageFullName)
+{
+    AutoPtr<MsixRequest> removePackageRequest;
+    RETURN_IF_FAILED(MsixRequest::Make(OperationType::Remove, Flags::NoFlags, std::wstring(), packageFullName, MSIX_VALIDATION_OPTION::MSIX_VALIDATION_OPTION_FULL, &removePackageRequest));
+
+    const HRESULT hrCancelRequest = removePackageRequest->ProcessRequest();
+    if (FAILED(hrCancelRequest))
+    {
+        TraceLoggingWrite(g_MsixTraceLoggingProvider,
+            "Failed to process cancel request",
+            TraceLoggingLevel(WINEVENT_LEVEL_WARNING),
+            TraceLoggingValue(hrCancelRequest, "HR"));
+    }

```

Comment:
you should be getting a compiler warning or error that there's a codepath that doesn't return an HRESULT? should be return S_OK at the end.

Reply:
Modified to return S_OK.